scripts/pythondistdeps.py

```diff
@@ -18,6 +18,96 @@
 from warnings import warn
 
 
+class RpmVersion():
+    def __init__(self, version_id):
+        version = parse_version(version_id)
+        if isinstance(version._version, str):
+            self.version = version._version
+        else:
+            self.epoch = version._version.epoch
+            self.version = list(version._version.release)
+            self.pre = version._version.pre
+            self.dev = version._version.dev
+            self.post = version._version.post
+
+    def increment(self):
+        self.version[-1] += 1
+        self.pre = None
+        self.dev = None
+        self.post = None
+        return self
+
+    def __str__(self):
+        if isinstance(self.version, str):
+            return self.version
+        if self.epoch:
+            rpm_epoch = str(self.epoch) + ':'
+        else:
+            rpm_epoch = ''
+        while self.version[-1] == 0:
+            self.version.pop()
+        rpm_version = '.'.join(str(x) for x in self.version)
+        if self.pre:
+            rpm_suffix = '~%s' % ''.join(str(x) for x in self.pre)
+        elif self.post:
+            rpm_suffix = '^post%d' % self.post[1]
+        else:
+            rpm_suffix = ''
+        return '%s%s%s' % (rpm_epoch, rpm_version, rpm_suffix)
+
+def convert_compatible(name, operator, version_id):
+    if version_id.endswith('.*'):
+        return 'Invalid version'
+    version = RpmVersion(version_id)
+    if len(version.version) == 1:
+        return 'Invalid version'
+    upper_version = RpmVersion(version_id)
+    upper_version.version.pop()
+    upper_version.increment()
+    return '(%s >= %s with %s < %s)' % (
+        name, version, name, upper_version)
+
+def convert_equal(name, operator, version_id):
+    if version_id.endswith('.*'):
+        version_id = version_id[:-2] + '.0'
+        return convert_compatible(name, '~=', version_id)
+    version = RpmVersion(version_id)
+    return '%s = %s' % (name, version)
+
+def convert_not_equal(name, operator, version_id):
+    if version_id.endswith('.*'):
+        version_id = version_id[:-2]
+        version = RpmVersion(version_id)
+        lower_version = RpmVersion(version_id).increment()
+    else:
+        version = RpmVersion(version_id)
+        lower_version = version
+    return '(%s < %s or %s > %s)' % (
+        name, version, name, lower_version)
+
+def convert_ordered(name, operator, version_id):
+    if version_id.endswith('.*'):
+        version_id = version_id[:-2]
+        version = RpmVersion(version_id)
+        if '>' == operator:
+            operator = '>='
+            version.increment()
```

Comment:
Does this mean that `> 2.5` is converted to `>= 2.6`? Is that how Python versions work, i.e. `> 2.5` means `2.5.1` is not good? (That actually makes sense, just making sure.)

Reply:
This is in the case that the version ends with `.*`.  Assuming that you mean `> 2.5.*`, yes, that would be converted to `>= 2.6`.  I want to look at the spec again, because I no longer remember on what I based that conversion and it seems wrong.  pip seems happy with 1.0.0 matching 1>1.0.*`

```
$ pip install 'cairocffi > 1.0.*,< 1.0.1'
Collecting cairocffi<1.0.1,>1.0.*
  Downloading https://files.pythonhosted.org/packages/8e/05/143647cb9c228122264a70f82543bbc9cc499b4078c242a24b6b546e2050/cairocffi-1.0.0-py3-none-any.whl
```

Reply:
I'm checking with the distutils-sig.  The current behavior of "> version.*" is equivalent to ">= version", which seems like it might not be intentional.

```
>>> pkg_resources._vendor.packaging.requirements.Requirement('cairocffi > 1.0.*,< 1.0.1').specifier._specs
frozenset({<LegacySpecifier('>1.0.*')>, <Specifier('<1.0.1')>})

>>> '1.7' in pkg_resources.Requirement.parse('foo>1.7.*')
True
```

Reply:
It's not intentional, and undefined.  I think this interpretation/conversion is fine, though.

https://mail.python.org/archives/list/distutils-sig@python.org/thread/NWEQVTCX5CR2RKW2LT4H77PJTEINSX7P/